src/store/modules/segmentationModule.js

```diff
@@ -0,0 +1,1411 @@
+import EVENTS from '../../events.js';
+
+import { getters as storeGetters } from '../index.js';
+
+import { getLogger } from '../../util/logger.js';
+import pointInImage from '../../util/pointInImage.js';
+
+const logger = getLogger('store:modules:segmentationModule');
+
+// Internal state of the brush module.
+const state = {
+  colorMapId: 'BrushColorMap',
+  series: {},
+  colorLutTables: {},
+};
+
+const configuration = {
+  renderOutline: true,
+  renderFill: true,
+  renderInactiveLabelmaps: true,
+  radius: 10,
+  minRadius: 1,
+  maxRadius: 50,
+  segmentsPerLabelmap: 65535, // Max is 65535 due to using 16-bit Unsigned ints.
+  fillAlpha: 0.2,
+  fillAlphaInactive: 0.1,
+  outlineAlpha: 0.7,
+  outlineAlphaInactive: 0.35,
+  outlineWidth: 3,
+};
+
+/**
+ * A map of `firstImageId` to associated `BrushStackState`, where
+ * `firstImageId` is the `imageId` of the first image in a stack.
+ *
+ * @typedef {Object} Series
+ */
+
+/**
+ * @typedef {Object} BrushStackState An object defining a set of 3D labelmaps
+ *    associated with a specific cornerstone stack.
+ * @property {string} activeLabelmapIndex The index of the active `Labelmap3D`.
+ * @property {Labelmap3D[]} labelmaps3D An array of `Labelmap3D` objects.
+ */
+
+/**
+ * An
+ *
+ * @typedef {Object} Labelmap3D An object defining a 3D labelmap.
+ * @property {ArrayBuffer}  buffer An array buffer to store the pixel data of the `Labelmap3D` (2 bytes/voxel).
+ * @property {Labelmap2D[]} labelmaps2D array of `labelmap2D` views on the `buffer`, indexed by in-stack
+ *                          image positions.
+ * @property {Object[]} metadata An array of metadata per segment. Metadata is optional and its form is
+ *                               application specific.
+ * @property {number} activeSegmentIndex The index of the active segment for this `Labelmap3D`.
+ * @property {boolean[]} segmentsVisible The visibility of segments on this labelmap.
+ *                                       If an element is undefined, the visibility of that defaults to true.
+ */
+
+/**
+ * @typedef {Object} Labelmap2D An object defining a 2D view on a section of a `Labelmap3D`'s `buffer`.
+ * @property {Uint16Array} pixelData A 2D view on a section of the parent `Labelmap3D`'s `buffer`.
+ * @property {number[]} segmentsOnLabelmap An array of segments present in the `pixelData`.
+ * @property {boolean} invalidated Whether the data has been changed recently, and a new `ImageBitmap` should be created
+ *                                 for the segmentation fill should this `labelmap2D` be rendered.
+ */
+
+/**
+ * GetMetadata - Returns the metadata object for a partiular segment if
+ * segmentIndex is specified, otherwise returns an array of all segment metadata
+ * for the labelmap.
+ *
+ * @param  {HTMLElement|string} elementOrEnabledElementUID   The cornerstone enabled
+ *                                                    element or its UUID.
```

Comment:
Is the formatter breaking JS Doc into new lines for you? Curious if intellisense/parser picks this up okay.